### ISSUE TITLE
Queries failing after 60s despite changing dataproxy.timeout to > 60s(changed from 30/default but 180 is not reflecting) for Influx datasource 

### ISSUE THREAD

**Issue by som-kanade** [grafana/grafana#29457]
What happened:
Long running queries from Influx Datasource failed to fetch for larger data within 60s but dataproxy.timeout is changed to 180s (which is not reflecting)

What you expected to happen:
We expect that queries should not appear to time out in less time than has been configured in the dataproxy.timeout setting.

**Environment**:
- Grafana version: 7.X.X
- Data source type & version: InfluxDB
- OS Grafana is installed on: Ubuntu
- User OS & Browser: Chrome
- Grafana plugins:
- Others:

**Comment by som-kanade** [comment]
any updates here ?

**Comment by mdicss** [comment]
I have the same problem.

**Comment by gabor** [comment]
there is an issue with `dataproxy.timeout` (https://github.com/grafana/grafana/issues/34177), it might cause this problem, so we'll have to wait until that one is fixed.

**Comment by gabor** [comment]
@som-kanade @mdicss i tried this in both grafana 7.5.7 and in current `main`-branch, and the connection is not cut after 60 seconds when the timeout is set to a large enough value.

is it possible that some network equipment between the influxdb database and grafana-server is dropping the connection after 60 seconds?

if not, could you please provide more detailed info? :
- version of grafana
- version of influxdb
- is grafana and influxdb running on the same computer? natively or in docker?
- are you using the InfluxQL query language or the Flux query language?
- how exactly is the fetch dropped at 60 seconds? what error message do you see? is it always exactly 60 seconds?

thanks.

**Comment by som-kanade** [comment]
Hey @gabor 
Grafana version > 7.0.3
Influx version > 1.7.9
is grafana and influxdb running on the same computer? natively or in docker? > No they are running on different EC2 boxes
are you using the InfluxQL query language or the Flux query language? > influx query 
how exactly is the fetch dropped at 60 seconds? what error message do you see? is it always exactly 60 seconds? > yes most of the times it times aafter 60s

thanks

**Comment by marefr** [comment]
Fixed by #34597

**Comment by gabor** [comment]
hi @som-kanade thanks for the info. please try this out with grafana8 beta 3 or later (when it is released, it is not released yet). there were changes to the timeout handling that could fix the problem. if it does not we can reopen the issue.

also, moving to a newer influxdb version is worth a try, the "1.x" series is at 1.8.6 currently.

another approach is to, for a test-case, move both grafana and influxdb to the same EC2 box and try it again, something between the boxes might cut the connection after 60seconds.

**Comment by marefr** [comment]
@rodolk flux support has been part of Grafana since v7 or something. Make sure you're using the builtin Influx datasource rather than any external plugin. The code in question for the builtin flux support is here https://github.com/grafana/grafana/blob/f1fb202284a961bd8e3229742e57cc834a201078/pkg/tsdb/influxdb/flux/flux.go#L77-L82

**Comment by rodolk** [comment]
@marefr OK, I assume I'm using the builtin influx datasource.
Is SetHttpRequestTimeout being called somewhere?

**Comment by marefr** [comment]
No, doesn't look like it. It's weird influx client provide a separate API for this and not reusing the existing [http.Client](https://pkg.go.dev/net/http#Client) Timeout field - then it would have just worked as expected. Maybe a newer version of the influx client handles this better?

**Comment by gabor** [comment]
@marefr it's strange, this is an old issue, but originally i did test it and larger-than-60-second timeouts worked fine. it was not possible to reproduce the problem. anyway, if the timeout-value is not propagated, then this is something we should look at (i'm still confused why it worked in the past). i'll reopen the issue, but i don't work in this area anymore, so pinging @grafana/observability-metrics 👍

**Comment by lars20070** [comment]
I face the same problem. I have connected Grafana to an InfluxDB data source.

I have extended the config `/etc/grafana/grafana.ini` as below and restarted the `grafana-server` service.
```
[dataproxy]
timeout = 1200
```
Yet if I run large queries in Grafana (which I expect to take several minutes) I get the timeout error below.
```
Post “[http://localhost:8086/api/v2/query?org=main-org”:](http://localhost:8086/api/v2/query?org=main-org%E2%80%9D:) net/http: timeout awaiting response headers (Client.Timeout exceeded while awaiting headers)
```
It seems the client i.e. Grafana runs into a timeout. But my config changes have no effect.

Grafana version: 10.2.2
